### PR TITLE
Integrated backend to follow-suggestions section and counters

### PR DESCRIPTION
**Pull request by jeromes123**
**OVERVIEW**

-  Created and integrated backend for suggested users to follow
-  Added relationship methods to user to get followed users
-  Added relationship methods to user to get followers
-  Implemented counters to user's posts count, following count, and followers count

### CODE REVIEW

File: src/app/Models/User.php [hunk 1]
```diff
@@ -53,6 +54,30 @@ public function profile(): HasOne
     {
         return $this->hasOne(Profile::class);
     }
+
+    /**
+     * Get the users who are followed by the user.
+     */
+    public function followedUsers()
```
Comment (hipe01mark) [inline src/app/Models/User.php]: return type!!!

File: src/app/Models/User.php [hunk 2]
```diff
@@ -53,6 +54,30 @@ public function profile(): HasOne
...
+    {
+        return $this->hasMany(Relationship::class, 'follower_id', 'id');
+    }
+
+    /**
+     * Get the users who are followed by the user.
+     */
+    public function followers():HasMany
```
Comment (hipe01mark) [inline src/app/Models/User.php]: too close with the ":"

File: src/app/Models/User.php [hunk 3]
```diff
@@ -53,6 +54,30 @@ public function profile(): HasOne
...
+    {
+        return $this->hasMany(Relationship::class, 'following_id', 'id');
+    }
+
+    /**
+     * Get the posts created by the user
+     */
+    public function userPost():HasMany
```
Comment (hipe01mark) [inline src/app/Models/User.php]: too close with the ":"

File: src/routes/web.php [hunk 4]
```diff
@@ -43,6 +43,8 @@
     Route::post('/add-comment', [PostCommentController::class, 'create'])->name('post.comment.create');
     Route::delete('/delete-comment', [PostCommentController::class, 'delete']);
     Route::put('/edit-comment', [PostCommentController::class, 'update']);
+    Route::put('/edit-comment', [PostCommentController::class, 'update']);
```
Comment (hipe01mark) [inline src/routes/web.php:46]: why other routes no route name?

Comment (jeromes123) [reply src/routes/web.php:46]: can we do the naming for the routes in another PR?

Comment (hipe01mark) [reply src/routes/web.php:46]: noted

**Review by hipe01mark — APPROVED** [review]
GEFN 🙇 

If you are implementing a web application, always assign route names and use them with route(), but if the project is focused on RESTFul API implementation, it may not be necessary.

**Comment by jeromes123** [comment]
> GEFN 🙇
> 
> If you are implementing a web application, always assign route names and use them with route(), but if the project is focused on RESTFul API implementation, it may not be necessary.

Noted, sir. Thank you

**Pull request merged**